### PR TITLE
Add MultiRowRangeFilter support 

### PR DESCRIPTION
**Pull request by igorbernstein2**
Depends on #1260 for the hbase upgrade.

### CODE REVIEW

File: bigtable-hbase-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/adapters/filters/MultiRowRangeFilterAdapter.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
+package com.google.cloud.bigtable.hbase.adapters.filters;
```
Comment (sduskis) [inline bigtable-hbase-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/adapters/filters/MultiRowRangeFilterAdapter.java]: we need the standard apache license header on all files.

File: bigtable-hbase-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/adapters/filters/MultiRowRangeFilterAdapter.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+/**
+ * Adapter for {@link MultiRowRangeFilter}, it converts the filter into an index scan hint
+ */
+public class MultiRowRangeFilterAdapter extends TypedFilterAdapterBase<MultiRowRangeFilter>  {
+  @Override
+  public RowFilter adapt(FilterAdapterContext context, MultiRowRangeFilter filter)
+      throws IOException {
+
+    return RowFilter.newBuilder()
```
Comment (sduskis) [inline bigtable-hbase-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/adapters/filters/MultiRowRangeFilterAdapter.java]: What happen in cases where scan hints won't be invoked, like in a must pass one filter list?

Comment (sduskis) [inline bigtable-hbase-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/adapters/filters/MultiRowRangeFilterAdapter.java]: We somehow need to error out in that case...

File: bigtable-hbase-parent/bigtable-hbase-integration-tests/src/test/java/com/google/cloud/bigtable/hbase/TestFilters.java [hunk 3]
```diff
@@ -1610,6 +1612,40 @@ public void testPrefixFilter() throws IOException {
       Assert.assertArrayEquals(rowKeys[i+1], boundedResults[i].getRow());
     }
   }
+  @Test
+  public void testMultiRangeFilter() throws IOException {
```
Comment (sduskis) [inline bigtable-hbase-parent/bigtable-hbase-integration-tests/src/test/java/com/google/cloud/bigtable/hbase/TestFilters.java]: What happens when the tests are set to HBase 1.0?

Comment (sduskis) [inline bigtable-hbase-parent/bigtable-hbase-integration-tests/src/test/java/com/google/cloud/bigtable/hbase/TestFilters.java]: Where's the shim?

**Comment by sduskis** [comment]
Overall, I'm concerned about this filter.  There are some corner cases that need to be handled, and there doesn't seem to be a good way to handle them now.  There are some other solutions that solve the same problem that this PR solves, specifically `BigtableExtendedScan`, which better aligns with the Cloud Bigtable model than this implementation, and doesn't have the same corner cases.

I think we should continue working on this, but not include this as part of the 0.9.6 release.

File: bigtable-hbase-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/adapters/filters/MultiRowRangeFilterAdapter.java (hunk 1 above)
Comment (igorbernstein2) [inline bigtable-hbase-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/adapters/filters/MultiRowRangeFilterAdapter.java]: done

File: bigtable-hbase-parent/bigtable-hbase-integration-tests/src/test/java/com/google/cloud/bigtable/hbase/TestFilters.java (hunk 3 above)
Comment (igorbernstein2) [inline bigtable-hbase-parent/bigtable-hbase-integration-tests/src/test/java/com/google/cloud/bigtable/hbase/TestFilters.java]: Oops, added the shim.

**Comment by sduskis** [comment]
I would feel more comfortable with this if this required explicit user action to enable `MultiRowRangeFilterAdapter.java`. 

I would approve this PR if the integration test were marked `KnownGap`, with a comment, and that there were instructions on `MultiRowRangeFilterAdapter` on how to enable it.

In a separate PR, let's make `FilterAdapter.addFilterAdapter()` public so that users can actually enable MultiRowRangeFilterAdapter.

File: bigtable-hbase-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/adapters/filters/MultiRowRangeFilterAdapter.java (hunk 2 above)
Comment (igorbernstein2) [inline bigtable-hbase-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/adapters/filters/MultiRowRangeFilterAdapter.java]: Good call, I added the limitation to isFilterSupported
